src/core/configuration/tree.ts

```diff
@@ -84,9 +84,17 @@ enum ValueType {
 function findValueInObject(obj: unknown, path: string[]): undefined | IValueDescriptor {
   const key = path.shift();
   if (!key) {
+    let type: ValueType;
+    if (obj instanceof Object && !(obj instanceof Array)) {
+      type = ValueType.Complex;
+    } else if (obj instanceof Array) {
+      type = ValueType.Array;
+    } else {
```

Comment:
We can simplify it. If we will check obj instanceof Array` first, then we will not need check `!(obj instanceof Array)`.